### PR TITLE
Updating CODEOWNERS for ACS Identity

### PR DESCRIPTION
**Pull request by petrsvihlik**
Reason: The ACS SDKs (including Identity) are being handed over to the respective service teams.

### CODE REVIEW

File: .github/CODEOWNERS
Comment (jsquire) [inline .github/CODEOWNERS:181]: Please be aware that this rule isn't likely to be triggered; to my knowledge, GitHub will match the first (loosest) rule and stop.   Since the top-level  `/sdk/communication` has an entry, I believe that will short-circuit and prevent L181 from triggering.    _(we're seeing this behavior with other services trying to do similar hierarchical routing)_

**Review by jsquire — APPROVED** [review]

File: .github/CODEOWNERS
Comment (RezaJooyandeh) [reply .github/CODEOWNERS:181]: Does it mean that the order should be reversed?

Comment (petrsvihlik) [reply .github/CODEOWNERS:181]: @jsquire Interesting, this [forum post](https://github.community/t/order-of-owners-in-codeowners-file/143073/2) suggest the opposite:

> specifically lines that come later override lines that appear earlier in the file
> ...
> this is taken advantage of by starting with the least specific rule and then following with increasingly more specific rules

And the code example in the [docs](https://docs.github.com/en/github/creating-cloning-and-archiving-repositories/creating-a-repository-on-github/about-code-owners) says basically the same.

Comment (jsquire) [reply .github/CODEOWNERS:181]: @petrsvihlik:  I don't disagree,  I'm just offering the observation based on other experiences.   For example, we tried using the following spread for Event Hubs, but continually see only the folks named in the top-level rule picked up.

```text
# PRLabel: %Event Hubs
/sdk/eventhub/                                  @serkantkaraca @jsquire

# PRLabel: %Event Hubs %Functions
/sdk/eventhub/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/ @pakrym @JoshLove-msft @jsquire

# ServiceLabel: %Event Hubs %Service Attention
/sdk/eventhub/Microsoft.Azure.EventHubs/   @serkantkaraca @sjkwak @kasun04

# ServiceLabel: %Event Hubs %Service Attention
/sdk/eventhub/Microsoft.Azure.EventHubs.Processor/  @serkantkaraca @sjkwak @kasun04

# ServiceLabel: %Event Hubs %Service Attention
/sdk/eventhub/Microsoft.Azure.EventHubs.ServiceFabricProcessor/  @JamesBirdsall @serkantkaraca @sjkwak
```

Comment (petrsvihlik) [reply .github/CODEOWNERS:181]: @jsquire right. Thanks for sharing the example. According to the documentation (`the last matching pattern takes the most
precedence`), reversing the order of the two lines won't help. So the only thing that comes to my mind is merging it the way we believe it should be and, if it won't work, raising an issue with the folks from GitHub.

Comment (weshaggard) [reply .github/CODEOWNERS:181]: I was also under the impression that it is supposed to take the last match in the file and I agree if we don't see that happening we should reach out to github to better understand why it isn't working like that.

Comment (weshaggard) [reply .github/CODEOWNERS:181]: I will also give a counter example which is we have a general rule at the top:

```
# Catch all
/sdk/                                           @AlexGhiondea
```

Which we then override further down the file and I'm pretty sure those overrides work. @jsquire are you seeing this only with code owners or with other things like the pipeline notifications. 

I will also add that I believe github does this match for every file in a PR so it might end up finding multiple set of owners to add.

Comment (weshaggard) [reply .github/CODEOWNERS:181]: Just as check after merging I verified on a file and this rule seems to work:

image

Comment (jsquire) [reply .github/CODEOWNERS:181]: @weshaggard:  We're seeing it specifically with CODEOWNERs for PRs.   The "Service Attention" bits are confirmed invalid, as the bot doesn't consider paths.  _(I need to leave myself a note to remove that duplicate entry)_

We've been seeing the same lack of respect for subpaths with the Service Bus configuration as well.

Comment (petrsvihlik) [reply .github/CODEOWNERS:181]: > Just as check after merging I verified on a file and this rule seems to work:
> 
> image

@weshaggard perfect, this appears to be fine but I discovered that the `/sdk/communication/ @acsdevx-msft` rule is not being applied. if I go to that folder (or any subfolder except Identity), I can't see the shield icon with code owners.

also, I got included in #22469 as a code owner by some magic :)...in the PR, there is no file that would fall under the `/sdk/communication/Azure.Communication.Identity/ @petrsvihlik @martinbarnas-ms` rule.

Do you have any idea regarding any of the issues, Wes?

Comment (weshaggard) [reply .github/CODEOWNERS:181]: For #22469 there was a commit https://github.com/Azure/azure-sdk-for-net/pull/22469/commits/29457e8500035ca5c1cde40cb985b36553af371c that touched a file under identity. I think it was later undone but it did touch that file in the commit chain which is why you were added.

As for communication directory they are using @acsdevx-msft which I thought was a team alias but I'm less sure about that now. However looking at https://github.com/acsdevx-msft I believe the issue is the account doesn't have contributor access to the repo so it isn't listed as an official owner. @meeraharidasa can you give more information on the @acsdevx-msft account you added to the code owners file with https://github.com/Azure/azure-sdk-for-net/pull/18806?

Comment (meeraharidasa) [reply .github/CODEOWNERS:181]: sure! This is the team mail alias that they (ACS) sent me for all the triaging to be sent to. Should this alias sign up for write access to our repos to avoid these issues?

Comment (weshaggard) [reply .github/CODEOWNERS:181]: @meeraharidasa lets continue this discussion in the PR that added it #18806

Comment (petrsvihlik) [reply .github/CODEOWNERS:181]: thanks @weshaggard ...yes the commit was undone. all clear now :)

**Pull request merged**
